From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Andrew Branson <andrew.branson@jolla.com>
Date: Thu, 6 Aug 2026 22:36:04 +0200
Subject: [sailfishos][webrtc] Add NSPR flags to camera target

The generated video-capture target clears the normal OS include flags,
while the Sailfish device-info source uses Gecko Preferences and therefore
NSPR types. Add the configured NSPR include path for geckocamera builds.

Allow WebRTC's reference-counting wrapper to subclass the Sailfish capture
implementation, matching the other platform capture implementations.
---
 .../libwebrtc/modules/video_capture/sfos/video_capture_sfos.h | 4 ++--
 .../video_capture/video_capture_internal_impl_gn/moz.build    | 1 +
 2 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/third_party/libwebrtc/modules/video_capture/sfos/video_capture_sfos.h b/third_party/libwebrtc/modules/video_capture/sfos/video_capture_sfos.h
index c9286b378dc63..3f43f9ff45b53 100644
--- a/third_party/libwebrtc/modules/video_capture/sfos/video_capture_sfos.h
+++ b/third_party/libwebrtc/modules/video_capture/sfos/video_capture_sfos.h
@@ -15,8 +15,8 @@
 namespace webrtc {
 namespace videocapturemodule {
 
-class VideoCaptureModuleSFOS final : public VideoCaptureImpl,
-                                     public gecko::camera::CameraListener {
+class VideoCaptureModuleSFOS : public VideoCaptureImpl,
+                               public gecko::camera::CameraListener {
  public:
   VideoCaptureModuleSFOS() = default;
   ~VideoCaptureModuleSFOS() override;
diff --git a/third_party/libwebrtc/modules/video_capture/video_capture_internal_impl_gn/moz.build b/third_party/libwebrtc/modules/video_capture/video_capture_internal_impl_gn/moz.build
index 305e2f109e472..9d2aa65020649 100644
--- a/third_party/libwebrtc/modules/video_capture/video_capture_internal_impl_gn/moz.build
+++ b/third_party/libwebrtc/modules/video_capture/video_capture_internal_impl_gn/moz.build
@@ -9,6 +9,7 @@ if CONFIG["MOZ_WIDGET_TOOLKIT"] == "gtk":
     CXXFLAGS += CONFIG["MOZ_GTK3_CFLAGS"]
 if CONFIG["MOZ_ENABLE_WEBRTC_GECKOCAMERA"]:
     CXXFLAGS += CONFIG["LIBGECKOCAMERA_CFLAGS"]
+    CXXFLAGS += CONFIG["NSPR_CFLAGS"]
     LOCAL_INCLUDES += [ "/media/libyuv/libyuv/include/" ]
     OS_LIBS += CONFIG["LIBGECKOCAMERA_LIBS"]
 else:
